pkg/server/compact.go

```diff
@@ -20,12 +21,16 @@ func isCompact(txn *etcdserverpb.TxnRequest) bool {
 func (l *LimitedServer) compact(ctx context.Context) (*etcdserverpb.TxnResponse, error) {
 	return &etcdserverpb.TxnResponse{
 		Header:    &etcdserverpb.ResponseHeader{},
-		Succeeded: true,
+		Succeeded: false,
```

Comment:
Note: this is an unrelated change that keeps the upstream k8s from calling compact. The compact call is also a noop on our side at the moment, but if we return failure here it won't even bother calling it, due to it thinking that another master has already compacted since the last time it tried.